frappe/model/naming.py

```diff
@@ -109,6 +108,8 @@ def make_autoname(key, doctype=''):
 			en = today.strftime("%d")
 		elif e=='YYYY':
 			en = today.strftime('%Y')
+		elif doc and doc.get(e,None):
```

Comment:
Don't need to pass 2nd arg as None. It could just be `doc and doc.get(e)`: en - doc.get(e)